# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: widget/nsBaseDragService.cpp
# Commit: 1332810e2604
# Full Hash: 1332810e2604ddb6982860f5a3f2412d2249b42c
# Author: Ryan Hunt <rhunt@eqrion.net>
# Date: 2019-05-23 15:55:11
# Regressor Bug: 1525720
# File Overlap Count: 1
# Description:
#   Bug 1525720, part 7 - Use getter methods to access mBrowserParent and mBrowserBridgeChild. r=farre
#   
#   This prepares nsFrameLoader for replacing mBrowserParent and mBrowserBridgeChild
#   with a common interface by making special case code use a getter method instead
#   of direct access.
# ==============================================================================

diff -r 21985ab71b5b -r 1332810e2604 widget/nsBaseDragService.cpp
--- a/widget/nsBaseDragService.cpp	Wed Apr 24 23:52:47 2019 -0500
+++ b/widget/nsBaseDragService.cpp	Thu Apr 25 12:36:16 2019 -0500
@@ -566,9 +566,8 @@
   if (flo) {
     RefPtr<nsFrameLoader> fl = flo->GetFrameLoader();
     if (fl) {
-      auto* tp =
-          static_cast<mozilla::dom::BrowserParent*>(fl->GetRemoteBrowser());
-      if (tp && tp->TakeDragVisualization(*aSurface, aScreenDragRect)) {
+      auto* bp = fl->GetBrowserParent();
+      if (bp && bp->TakeDragVisualization(*aSurface, aScreenDragRect)) {
         if (mImage) {
           // Just clear the surface if chrome has overridden it with an image.
           *aSurface = nullptr;
